mbuild/formats/gomc_conf_writer.py

```diff
@@ -4464,6 +4728,655 @@ def __init__(
+                                                "subvolumecenter"
+                                            ]:
+                                                if not isinstance(
+                                                    x_y_z_dim_i, int
+                                                ) and not isinstance(
+                                                    x_y_z_dim_i, float
+                                                ):
+                                                    bad_input_variables_values_list.append(
+                                                        "subvolumecenter"
+                                                    )
+                                        else:
+                                            bad_input_variables_values_list.append(
+                                                "subvolumecenter"
+                                            )
+                                    else:
+                                        bad_input_variables_values_list.append(
+                                            "subvolumecenter"
+                                        )
+
+                                # check for the SubVolumeCenterList if its a dynamic box
+                                elif target_swap_tag_id_dict_key_data[
+                                    "subvolumetype"
+                                ].lower() in ["dynamic"]:
+                                    if (
+                                        "subvolumecenterlist"
+                                        in target_swap_tag_id_dict_key_data
+                                    ):
+                                        if isinstance(
+                                            target_swap_tag_id_dict_key_data[
+                                                "subvolumecenterlist"
+                                            ],
+                                            list,
+                                        ):
+                                            for (
+                                                atom_index_dim_i
+                                            ) in target_swap_tag_id_dict_key_data[
+                                                "subvolumecenterlist"
+                                            ]:
+                                                if isinstance(
+                                                    atom_index_dim_i, int
+                                                ):
+                                                    if atom_index_dim_i < 0:
+                                                        bad_input_variables_values_list.append(
+                                                            "subvolumecenterlist"
+                                                        )
+                                                elif isinstance(
+                                                    atom_index_dim_i, str
```

Comment:
Yeah, I agree this may be cleanup up a bit in the future.  For now however, since it is working, I would like to focus on getting MoSDeF-GMSO-GOMC working, before making structural changes to the existing code(s).